habitat_baselines/rl/ppo/policy.py

```diff
@@ -63,10 +71,12 @@ class Net(nn.Module):
     goal vector with CNN's output and passes that through RNN.
     """
 
-    def __init__(self, observation_space, hidden_size):
+    def __init__(self, observation_space, hidden_size, goal_sensor_uuid):
```

Comment:
What a "goal_sensor" is is very task specific.  Different task will have wildly different ideas of a goal sensor (i.e. EQA has questions).  The network here should be a default network that doesn't have an idea of a goal and then you make a PointNavNetwork by inheriting from the Net and implementing something like `get_goal_embedding`.  Now instead of `x =  observations[self.goal_sensor_uuid]` you have something like `x = self.get_goal_embedding(observations)`.  That would just return `observations['pointgoal']` for PointNav, but allows something like EQA or ObjectNav to be done with minimal changes.

Reply:
@erikwijmans, agree with all your points. Maybe, we can make it as a next step, as a the scope of this task was more narrow. @JasonJiazhiZhang, let's reflect it in PR title. Something, like `Made PPO less tangled with pointnav task."